client/src/app/services/wordList.service.ts

```diff
@@ -33,4 +35,41 @@ export class WordListService {
       });
   }
 
+  updateWordInDataStore(word: Word){
+    this.dataStore.wordList = this.dataStore.wordList
+      .map(item => (item.id == word.id) ? word : item);
+
+    this.dataStore.wordList
+      .filter(item => {
+        if(item.id == word.id) this.dataStashToUpDate.push(item)
+      });
+
+    this.wordList.next(Object.assign({}, this.dataStore).wordList);
+  }
+
+  removeWordInDataStore(id: string){
+    this.dataStore.wordList
+      .filter(item => {
+        if(item.id == id) this.dataStashToRemove.push(id)
+      });
+
+    this.dataStore.wordList = this.dataStore.wordList
+      .filter(item => item.id != id);
```

Comment:
Wydaje mi się że filter jest tutaj lepsze jednak. A jeżeli chcesz tylko jedną wartość znaleźć to można też użyć '.find()', które działa tak jak filter, tylko zwraca jedną wartość, a nie tablice jak '.filter()'